src/OpenSage.Game/Logic/Object/Update/AIUpdate/AIUpdate.cs

```diff
@@ -98,7 +98,7 @@ internal AIUpdate(GameObject gameObject, GameContext context, AIUpdateModuleData
 
             SetLocomotor(LocomotorSetType.Normal);
 
-            if (ModuleData.Turret != null)
+            if (ModuleData != null && ModuleData.Turret != null)
```

Comment:
When did you find that `ModuleData` is `null`? I don't think that should ever be the case.

Reply:
I have actually run into this same issue locally recently when using the sav2json tool. I never paid much attention to what caused it...

Reply:
It just raised `null` error for me locally all the time and I haven't checked why this happens. Most likely this isn't proper fix, so I removed this change from PR and may try investigate it deeper some later.

Reply:
@tgjones It happens on loading ShellMap in Generals for me. `NullReferenceException` happens on execution of this [code](https://github.com/OpenSAGE/OpenSAGE/blob/490d996c320278833243773da6012adeb33cfc47/src/OpenSage.Game/Logic/Object/Update/AIUpdate/JetAIUpdate.cs#L420) in JetAIUpdate.cs. It triggers `AIUpdate` constructor and [this assignment](https://github.com/OpenSAGE/OpenSAGE/blob/490d996c320278833243773da6012adeb33cfc47/src/OpenSage.Game/Logic/Object/Update/AIUpdate/AIUpdate.cs#L94) doesn't work. `moduleData` is an instance of `JetAIUpdateModuleData` but `ModuleData` remains `null` after the assignment.

Reply:
Ah yup, that's it, and I have it fixed locally as part of the jet state stuff I'm working on.